src/src/cpp/environments/classic_control/Pendulum.h

```diff
@@ -4,202 +4,193 @@
-    double minActionContinuous() const { return -maxTorque; }
+    double theta() { return internal_state_[kThetaIndex]; }
+    double thetaDot() { return internal_state_[kThetaDotIndex]; }
 
     Pendulum() {
         n_eval_train_ = 20;
         n_eval_validation_ = 0;
         n_eval_test_ = 100;
-        disReset = uniform_real_distribution<>(-M_PI, M_PI);
-        disResetDot = uniform_real_distribution<>(-1.0, 1.0);
-        actionsDiscrete.push_back(-maxTorque);
+        disReset = std::uniform_real_distribution<>(-M_PI, M_PI);
+        reset_dot_distribution_ = std::uniform_real_distribution<>(-1.0, 1.0);
+        actionsDiscrete.push_back(-kMaxTorque);
         actionsDiscrete.push_back(0.0);
-        actionsDiscrete.push_back(maxTorque);
+        actionsDiscrete.push_back(kMaxTorque);
         eval_type_ = "Control";
         max_step_ = 300;
-        internal_state_.reserve(PENDULUM_DIM);
-        internal_state_.resize(PENDULUM_DIM);
-
-        max_costs =
-            pow(M_PI, 2) + 0.1 * pow(maxSpeed, 2) + 0.001 * pow(maxTorque, 2);
-        // max_costs_all = -(max_costs * max_step);
-        state_.reserve(PENDULUM_STATE_SIZE);
-        state_.resize(PENDULUM_STATE_SIZE);
-        state_po_.reserve(PENDULUM_STATE_SIZE - 1);
-        state_po_.resize(PENDULUM_STATE_SIZE - 1);
+        internal_state_.reserve(kPendulumDimensions);
+        internal_state_.resize(kPendulumDimensions);
+
+        max_costs_ = pow(M_PI, 2) + 0.1 * pow(kMaxSpeed, 2) + 
+                    0.001 * pow(kMaxTorque, 2);
+        state_.reserve(kPendulumStateSize);
+        state_.resize(kPendulumStateSize);
+        state_po_.reserve(kPendulumStateSize - 1);
+        state_po_.resize(kPendulumStateSize - 1);
     }
 
-    ~Pendulum() {}
-
-    void reset(mt19937 &rng) {
-        internal_state_[_theta] = disReset(rng);
-        internal_state_[_thetaDot] = disResetDot(rng);
+    void reset(std::mt19937& rng) override {
+        internal_state_[kThetaIndex] = disReset(rng);
+        internal_state_[kThetaDotIndex] = reset_dot_distribution_(rng);
 
-        state_[0] = state_po_[0] = cos(internal_state_[_theta]);
-        state_[1] = state_po_[1] = sin(internal_state_[_theta]);
-
-        state_[2] = internal_state_[_thetaDot];
-
-        // state[3] = disNoise(rng);
+        state_[0] = state_po_[0] = cos(internal_state_[kThetaIndex]);
+        state_[1] = state_po_[1] = sin(internal_state_[kThetaIndex]);
+        state_[2] = internal_state_[kThetaDotIndex];
 
         reward = 0;
-
         step_ = 0;
         terminalState = false;
     }
 
-    bool terminal() {
-        terminalState = step_ >= max_step_ ? true : false;
+    bool terminal() override {
+        terminalState = step_ >= max_step_;
         return terminalState;
     }
 
-    Results update(int actionD, double actionC, mt19937 &rng) {
-        (void)actionD;
-        double torque = bound(actionC, -maxTorque, maxTorque);
+    Results update(int action_discrete, double action_continuous, 
```

Comment:
Good work overall. I think that you should make the functions CamelCase tho for the 3 files, make sure the functions are capitalized too

Reply:
fyi some functions are implemented like this is inherited from TaskEnv, so we cant really change the capitalization.